README.md

```diff
@@ -1,86 +1,149 @@
+This builds dependencies in the folder "depend".
 
+Note that if rocBLAS and MIOpen are installed with the command ```sudo apt install -y rocblas miopen-hip``` as mentioned
+above, we can comment out the two lines 
+```
+ROCmSoftwarePlatform/rocBLAS@abd98a2b48b29326ebaef471630786a548622c06
+ROCmSoftwarePlatform/MIOpen@2.4.0
+```
+in the file `requirements.txt` (adding a '#' character at the start of each line).
+Also note that for ROCm3.7 and later release, Ubuntu 18.04 or later releases are needed. Upgrapde to Ubuntu 18.04 can be
+done as:
 
 ```
-CXX=/opt/rocm/llvm/bin/clang++ cmake ..
+sudo apt update
+sudo apt install linux-headers-4.18.0-25-generic linux-image-4.18.0-25-generic  linux-modules-4.18.0-25-generic linux-modules-extra-4.18.0-25-generic -y
 ```
 
-If the dependencies from `install_deps.cmake` was installed to another directory, the `CMAKE_PREFIX_PATH` needs to be set to what `--prefix` was set to from `install_deps.cmake`:
+### Building MIGraphX source
+
+First create a build directory:
 
 
 ```
-CXX=/opt/rocm/llvm/bin/clang++ cmake -DCMAKE_PREFIX_PATH=/some/dir ..
+mkdir build; 
+cd build;
 ```
 
+Next, configure the cmake. If the script [buiild_prereqs.sh](./tools/build_prereqs.sh) is used to install the
+dependencies, then all dependencies are installed at default locations, and MIGraphX can be build as:
+* In ROCm3.3:
 
-#### Changing the cmake configuration
+```
+CXX=/opt/rocm/bin/hcc cmake ..
+```
 
-The configuration can be changed after running cmake by using `ccmake`:
+* In ROCm3.7 or later releases:
 
-` ccmake .. ` **OR** `cmake-gui`: ` cmake-gui ..`
+```
+CXX=/opt/rocm/llvm/bin/clang++ cmake ..
+```
 
-## Building the library
+If the above rbuild command was used to build and install the dependencies, then all dependencies are in the folder
+"depend", and the `CMAKE_PREFIX_PATH` needs to be set to the same folder "depend" (full path are needed here.), and the
+command is:
 
-The library can be built, from the `build` directory using the 'Release' configuration:
+* ROCM3.3:
 
-` cmake --build . --config Release ` **OR** ` make `
+```
+CXX=/opt/rocm/bin/hcc cmake -DCMAKE_PREFIX_PATH=../depend ..
 
-And can be installed by using the 'install' target:
+```
+* ROCM3.7 or later releases:
 
-` cmake --build . --config Release --target install ` **OR** ` make install `
+```
+CXX=/opt/rocm/llvm/bin/clang++ cmake -DCMAKE_PREFIX_PATH=../depend ..
+```
 
-This will install the library to the `CMAKE_INSTALL_PREFIX` path that was set.
+Then we can build MIGraphX source code as:
 
-To build a debug version of the library, the cmake variable `CMAKE_BUILD_TYPE` can be set to `Debug`.
+```
+make -j$(nproc)
+```
 
-` cmake -DCMAKE_BUILD_TYPE=Debug . `
+and correctness can be verified as:
 
-## Running the tests
+```
+make -j$(nproc) check
+```
 
-The tests can be run by using the 'check' target:
+MIGraphX libs can be installed as:
 
-` cmake --build . --config Release --target check ` **OR** ` make check `
+```
+sudo make install
```

Comment:
Dont put `sudo` here. If the user sets the install prefix to a local path then sudo is nto necessary.

Reply:
sudo is removed.